packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -148,64 +155,109 @@ export function toVisualTree(tree: VisualSharedTreeNode): VisualChildNode {
 }
 
 /**
- * Concatenrate allowed types for `ObjectNodeStoredSchema` and `MapNodeStoredSchema`.
+ * Concatenate allowed types for `ObjectNodeStoredSchema` and `MapNodeStoredSchema`.
  */
-function concatenateTypes(fieldTypes: ReadonlySet<string>): string {
+export function concatenateTypes(fieldTypes: ReadonlySet<string>): string {
 	return [...fieldTypes].join(" | ");
 }
 
 /**
- * Returns the allowed fields & types for the object fields (e.g., `foo : string | number, bar: boolean`)
+ * Properties that describe schema constraints for a field in the tree
  */
-function getObjectAllowedTypes(schema: SimpleObjectNodeSchema): string {
-	const result: string[] = [];
+interface FieldSchemaProperties {
+	/**
+	 * Set of type names that are valid for this specific node position in the tree.
+	 * This is a subset of the types defined in treeDefinitions.
+	 */
+	allowedTypes: ReadonlySet<string> | undefined;
 
-	for (const [fieldKey, treeFieldSimpleSchema] of Object.entries(schema.fields)) {
-		const fieldTypes = treeFieldSimpleSchema.allowedTypes;
-		result.push(`${fieldKey} : ${concatenateTypes(fieldTypes)}`);
-	}
-
-	return `{ ${result.join(", ")} }`;
+	/**
+	 * Whether this field is required in its parent object schema.
+	 * Only meaningful for direct children of object nodes.
+	 * Undefined for array/map elements since they are always required within their parent.
```

Comment:
Not sure about this line, specifically "since they are always required within their parent". I don't recall what we do at the schema level with those "dynamic fields" inside arrays/maps (@Josmithr, maybe you remember?), but at least the way we're phrasing it here seems like it might be incorrect. My suggestion warrants double-checking for accuracy as well, thought.
```suggestion
	 * Whether this field is required in its parent object schema.
	 * Only meaningful for fields of object nodes.
	 * Fields of array/map nodes are treated differently, which is reflected here by using `undefined`.
```

Reply:
I think the description is off a bit in terms of layering. A field will always exist in the parent as defined. It's the data _within_ the field that is potentially optional.

Another way of putting it is that the field determines whether or not its child data is required or optional (or an identifier). 

Since we are documenting a property of a _field_ here, I think we could simplify the description.
```suggestion
	 * Whether the field is required (true) or optional (false).
	 *
	 * `undefined` indicates that the field is implicitly required.
	 * In this case, no requirement information will be displayed by the devtools.
```